treesit-auto.el

```diff
@@ -147,22 +197,12 @@ Returns `non-nil' if install was completed without error."
 If the tree-sitter grammar is missing for the current major mode,
 it will prompt the user if they want to install it from the
 currently registered repository.  If the user chooses to install
-the grammar it will then re-enable the current major-mode."
-  (when-let* ((mode (symbol-name major-mode))
-              (lang (and (string-match "\\(.*\\)-ts-mode$" mode)
-                         (intern (replace-regexp-in-string
-                                  "\\(.*\\)-ts-mode$" "\\1"
-                                  mode))))
-              ((not (treesit-ready-p lang 't)))
-              ((and
-                treesit-auto-install
-                (treesit-auto--prompt-to-install-package lang))))
-    ;; We need to rerun the current major mode after a successful
-    ;; install because we only hook into after the major-mode has
-    ;; finished setup. So, if the install fails it will fail to load
-    ;; or fallback to the mode defined in the remap-alist. But, if it
-    ;; succeeds we assume the user wants to use the `ts-mode'.
-    (funcall major-mode)))
+the grammar it will then switch to the tree-sitter powered
+version of the current major-mode."
+  (when-let* ((not-ready (not (treesit-auto--ready-p major-mode)))
+              (lang (treesit-auto--lang major-mode))
+              (install-success (treesit-auto--prompt-to-install-package lang)))
+    (funcall (intern (concat (symbol-name lang) "-ts-mode")))))
```

Comment:
@justinbarclay This is one line I was having trouble with that you might be interested in re-testing?  

When using `(funcall major-mode)` along with the changes to the `when-let*` block, it was no longer activating the tree-sitter version for me.  Instead, I was able to get the desired behavior by calling the corresponding tree-sitter mode like this.

The original comment mentioned "fail to load or fallback", but I wasn't able to reproduce that behavior.  For instance, I could force installation to fail by disabling internet or malforming a URL, yet I would still wind up back in go-mode, python-mode, or js-mode.

Reply:
I've been playing with this a bit this afternoon, and it runs great. 

> So, if the install fails, it will fail to load or fall back to the mode defined in the remap-alist. But, if it
I guess this comment was poorly written. There were two behaviours that would occur:
1. the `ts-mode` would fail to load 
   - in my case this was with `typescript` or `tsx` mode
2. the second behaviour is that it would fall back to the mode outlined in the remap-alist. 

However, seeing that we just installed the grammar, the behaviour we probably wanted was to restart the `ts-mode`.